animations/lib/src/basics/08_animated_list.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class AnimatedListDemo extends StatefulWidget {
+  static String routeName = '/basics/08_animated_list';
+
+  _AnimatedListDemoState createState() => _AnimatedListDemoState();
+}
+
+class _AnimatedListDemoState extends State<AnimatedListDemo> {
+  final GlobalKey<AnimatedListState> _listKey = GlobalKey();
+
+  void addUser() {
+    int index = listData.length;
+    listData.add(listData[Random().nextInt(5)]);
+    _listKey.currentState
+        .insertItem(index, duration: Duration(milliseconds: 500));
+  }
+
+  void deleteUser(int index) {
+    var user = listData.removeAt(index);
+    _listKey.currentState.removeItem(
+      index,
+      (context, animation) {
+        return FadeTransition(
+          opacity:
+              CurvedAnimation(parent: animation, curve: Interval(0.5, 1.0)),
+          child: SizeTransition(
+            sizeFactor:
+                CurvedAnimation(parent: animation, curve: Interval(0.0, 1.0)),
+            axisAlignment: 0.0,
+            child: _buildItem(user),
+          ),
+        );
+      },
+      duration: Duration(milliseconds: 600),
+    );
+  }
+
+  Widget _buildItem(UserModel user, [int index]) {
+    return ListTile(
+      key: ValueKey<UserModel>(user),
+      title: Text(user.firstName),
+      subtitle: Text(user.lastName),
+      leading: CircleAvatar(
+        backgroundImage: NetworkImage(user.profileImageUrl),
+      ),
+      onLongPress: index != null ? () => deleteUser(index) : null,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Animated List Demo"),
+        centerTitle: true,
+        leading: IconButton(
```

Comment:
This icon button is appear in the place normally used for the AppBar's back button. Rather than putting it into the `leading` property, add it to the AppBar's `actions` list. That way it'll appear on the right side, and the back button will still be present as in the other demos.